pilot/pkg/networking/core/route/route.go

```diff
@@ -1387,9 +1387,14 @@ func TranslateFault(in *networking.HTTPFaultInjection) *xdshttpfault.HTTPFault {
 
 func TranslateRequestMirrorPolicy(dst *networking.Destination, service *model.Service,
 	listenerPort int, mp *core.RuntimeFractionalPercent,
+) *route.RouteAction_RequestMirrorPolicy {
+	return TranslateRequestMirrorPolicyCluster(GetDestinationCluster(dst, service, listenerPort), mp)
+}
+
+func TranslateRequestMirrorPolicyCluster(cluster string, mp *core.RuntimeFractionalPercent,
```

Comment:
why abstact another function

Reply:
what do you prefer? to just copy the creation into the 3 places we use it?

Reply:
Maybe i misread it, cannot `TranslateRequestMirrorPolicy` be used

Reply:
Actually i donot see the diff with/without this patch

Reply:
Yeah this is extremely confusing

Reply:
I renamed stuff to be `waypointXYZ`